### ISSUE TITLE
Compile SDK 33 Deprecation Warnings

### ISSUE THREAD

**Issue by TalhaR** [raamcosta/compose-destinations#262]
Hey,

My team recently set our compile sdk to API 33 and had some warnings regarding deprecated code in some generated files. We are suppressing the warnings for now but it would be nice to have them fixed. Let me know if you need any more information from me or if there's anything I could do to help. Thanks! 😃 

```kotlin
w: .../build/generated/ksp/.../navtype/ErrorScreenDataNavType.kt: (22, 23): 'getSerializable(String?): Serializable?' is deprecated. Deprecated in Java
w: .../build/generated/ksp/.../navtype/ErrorScreenDataNavType.kt: (46, 45): 'getSerializable(String?): Serializable?' is deprecated. Deprecated in Java
w: .../build/generated/ksp/.../navtype/FakeFinancialCategoriesGET200SuccessDataNavType.kt: (22, 23): 'getSerializable(String?): Serializable?' is deprecated. Deprecated in Java
w: .../build/generated/ksp/.../navtype/FakeFinancialCategoriesGET200SuccessDataNavType.kt: (46, 45): 'getSerializable(String?): Serializable?' is deprecated. Deprecated in Java
```

We fixed a similar issue by adding a Compat function like this 
```kotlin
inline fun <reified T : Serializable> Bundle.getSerializableCompat(key: String): T? {
    return if (VERSION.SDK_INT >= VERSION_CODES.TIRAMISU) {
        this.getSerializable(key, T::class.java)
    } else {
        @Suppress("DEPRECATION")
        this.getSerializable(key) as? T
    }
}
```

**Comment by raamcosta** [comment]
Hi @TalhaR !

What version are you guys using? I added suppression on the latest version.

**Comment by TalhaR** [comment]
Hey @raamcosta we're using version `1.5.5-beta`. We haven't updated to the latest version because we're also using an older compose version. Is it possible to add the suppression to the older releases, maybe `1.6.22` or `1.5.22`? 

If not totally cool. I will try to update our version. Thanks for your help!

**Comment by raamcosta** [comment]
To be honest I kind of made a lazy call here 😁 I thought (if people are in latest sdk version, probably they can also update compose and Kotlin).

I know this is not necessarily true, but since we’re talking about warnings, I thought why not avoid some branch merging and stuff ahah.


So at the moment this is only fixed on 1.7.22-beta.

**Comment by TalhaR** [comment]
Yea totally understand your reasoning. Will try to update to that version. Closing this issue out. Thanks!
